rcl/test/rcl/test_memory.cpp

```diff
@@ -0,0 +1,330 @@
+              << "] - [ Msg : [ int64_value : " << pr.msg->int64_value
+              << " - string.value.size : " << pr.msg->string_value.size
+            << " ] ]" << '\n';
+}
+
+std::shared_ptr<test_msgs__msg__Primitives> getMessageWithStringLength(int length)
+{
+  std::shared_ptr<test_msgs__msg__Primitives> msg(
+    new test_msgs__msg__Primitives, [](test_msgs__msg__Primitives * ptr)
+    {
+      test_msgs__msg__Primitives__fini(ptr);
+      delete (ptr);
+    }
+  );
+
+  test_msgs__msg__Primitives__init(msg.get());
+  std::string test_str(length, 'x');
+  rosidl_generator_c__String__assign(&msg->string_value, test_str.c_str());
+
+  return msg;
+}
+
+std::shared_ptr<test_msgs__msg__Primitives> getMessageWithInt64Value(int64_t val)
+{
+  std::shared_ptr<test_msgs__msg__Primitives> msg(
+    new test_msgs__msg__Primitives, [](test_msgs__msg__Primitives * ptr)
+    {
+      test_msgs__msg__Primitives__fini(ptr);
+      delete (ptr);
+    }
+  );
+
+  test_msgs__msg__Primitives__init(msg.get());
+  msg->int64_value = val;
+
+  return msg;
+}
+
+class TEST_FIXTURE_P_RMW (TestMemoryFixture)
+  : public ::testing::TestWithParam<TestMemoryParams>
+{
+public:
+  rcl_context_t * context_ptr;
+  rcl_node_t * node_ptr;
+  void SetUp() override
+  {
+    rcl_ret_t ret;
+    {
+      rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+      ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+      OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+        EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options)) <<
+          rcl_get_error_string().str;
+      });
+      this->context_ptr = new rcl_context_t;
+      *this->context_ptr = rcl_get_zero_initialized_context();
+      ret = rcl_init(0, nullptr, &init_options, this->context_ptr);
+      ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    }
+    this->node_ptr = new rcl_node_t;
+    *this->node_ptr = rcl_get_zero_initialized_node();
+    const char * name = "test_publisher_node";
+    rcl_node_options_t node_options = rcl_node_get_default_options();
+    ret = rcl_node_init(this->node_ptr, name, "", this->context_ptr, &node_options);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  }
+
+  void TearDown() override
+  {
+    rcl_ret_t ret = rcl_node_fini(this->node_ptr);
+    delete this->node_ptr;
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    ret = rcl_shutdown(this->context_ptr);
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    ret = rcl_context_fini(this->context_ptr);
+    delete this->context_ptr;
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  }
+};
+
+/*
+  Memory test of publisher
+*/
+TEST_P_RMW(TestMemoryFixture, test_memory_publisher) {
+  osrf_testing_tools_cpp::memory_tools::ScopedQuickstartGtest scoped_quickstart_gtest(true);
+
+  auto common = [](auto & service) {service.print_backtrace();};
```

Comment:
The line two above should cause them all to print stack traces, right? So why have a custom one here? To avoid the gtest failure when they do so?